workspace/core/server/src/graphql/content/metadata_mutation.rs

```diff
@@ -71,6 +73,213 @@ impl MetadataMutationObject {
         Ok(new_metadata.into())
     }
 
+    async fn add_comment(
+        &self,
+        ctx: &Context<'_>,
+        metadata_id: String,
+        metadata_version: i32,
+        comment: CommentInput,
+    ) -> Result<i64, Error> {
+        let ctx = ctx.data::<BoscaContext>()?;
+        if ctx.principal.anonymous {
+            return Err(Error::new("unauthorized"));
+        }
+        let metadata_id = Uuid::parse_str(&metadata_id)?;
+        let check = PermissionCheck::new_with_metadata_id_with_version(
+            metadata_id,
+            metadata_version,
+            PermissionAction::View,
+        );
+        let mut metadata = ctx.metadata_permission_check(check).await?;
+        if !metadata.comments_enabled {
+            let check = PermissionCheck::new_with_metadata(
+                metadata,
+                PermissionAction::Manage,
+            );
+            metadata = ctx.metadata_permission_check(check).await?;
+        }
+        if comment.parent_id.is_none() && !metadata.comment_replies_enabled {
```

Comment:
Is this inverted?


 The intended behavior should be:
  - Block REPLIES when comment_replies_enabled = false
  - A reply is when parent_id.is_some() (has a parent)

  Correct code should be:
```
  if comment.parent_id.is_some() && !metadata.comment_replies_enabled {
      // This IS a reply, but replies are disabled
      // Require manager permission to create it anyway
      let check = PermissionCheck::new_with_metadata(metadata,
  PermissionAction::Manage);
      metadata = ctx.metadata_permission_check(check).await?;
  }
```
  The current code blocks top-level comments when threading is disabled, which is
  backwards - you'd want to allow top-level but block nested replies.